sdk/src/params.ts

```diff
@@ -0,0 +1,64 @@
+import type { EmbedOptions, OpenOptions } from './interfaces';
+
+type Options = Omit<OpenOptions & EmbedOptions, 'origin' | 'newWindow' | 'height' | 'width'>;
+
+const generators: Record<keyof Options, (value: any) => string> = {
+  clickToLoad: (value: Options['clickToLoad']) => trueParam('ctl', value),
+  devToolsHeight: (value: Options['devToolsHeight']) => percentParam('devToolsHeight', value),
+  forceEmbedLayout: (value: Options['forceEmbedLayout']) => trueParam('embed', value),
+  hideDevTools: (value: Options['hideDevTools']) => trueParam('hideDevTools', value),
+  hideExplorer: (value: Options['hideExplorer']) => trueParam('hideExplorer', value),
+  hideNavigation: (value: Options['hideNavigation']) => trueParam('hideNavigation', value),
+  showSidebar: (value: Options['showSidebar']) => booleanParam('showSidebar', value),
+  openFile: (value: Options['openFile']) => stringParams('file', value).join('&'),
+  theme: (value: Options['theme']) => enumParam('theme', ['light', 'dark'], value),
+  view: (value: Options['view']) => enumParam('view', ['preview', 'editor'], value),
+};
```

Comment:
Refactored how we generate the URL query string.

This is a bit more indirect than what we had before (a single function going through each option with code like `if (options.someOption) params.push('someOption='+options.someOption)`), but it has two upsides:

1. We can reuse code used to validate parameter values depending on their type.
2. Using the `Record<keyof Options, …>` type, we make sure that we are handling all supported options and not forgetting one. When we add an option, TS forces us to add a generator _or_ exclude that option name  in `
type Options = Omit<…>`.